homeassistant/components/comfoconnect/__init__.py

```diff
@@ -71,20 +96,49 @@ def setup(hass, config):
     hass.data[DOMAIN] = ccb
 
     # Start connection with bridge
-    ccb.connect()
+    await ccb.connect()
 
     # Schedule disconnect on shutdown
-    def _shutdown(_event):
-        ccb.disconnect()
-
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, _shutdown)
+    async def _shutdown(_event):
+        await ccb.disconnect()
 
-    # Load platforms
-    discovery.load_platform(hass, "fan", DOMAIN, {}, config)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _shutdown)
```

Comment:
From looking at the `cbb.disconnect` method it looks like  it will not work if you call it more than once.
If the entry is unloaded and HA is stopped this will be called twice, it to prevent it wrap it with `entry.async_on_unload`